lxd/config/map.go

```diff
@@ -95,14 +96,21 @@ func (m *Map) Change(changes map[string]interface{}) (map[string]string, error)
 func (m *Map) Dump() map[string]interface{} {
 	values := map[string]interface{}{}
 
-	for name, key := range m.schema {
-		value := m.GetRaw(name)
-		if value != key.Default {
-			if key.Hidden {
-				values[name] = true
-			} else {
-				values[name] = value
+	for name, value := range m.values {
+		key, ok := m.schema[name]
+		if ok {
+			// Schema key
+			value := m.GetRaw(name)
+			if value != key.Default {
+				if key.Hidden {
+					values[name] = true
+				} else {
+					values[name] = value
+				}
 			}
+		} else if strings.HasPrefix(name, "user.") {
```

Comment:
It feels like we should have a `IsUserConfig(name string)` function in `shared/` that we can then use eveywhere we do that kind of check.